slice-mapper/src/main/java/com/cognifide/slice/mapper/impl/processor/ChildrenFieldProcessor.java

```diff
@@ -67,9 +72,21 @@ public Object mapResourceToField(Resource resource, ValueMap valueMap, Field fie
 		final Class<?> fieldType = field.getType();
 		if (fieldType.isArray()) {
 			return getArrayFromList(fieldType.getComponentType(), mappedModels);
-		} else {
-			return mappedModels;
 		}
+		return getMappedModelsCollection(mappedModels, fieldType);
+	}
+
+	private Collection getMappedModelsCollection(List<?> mappedModels, Class<?> fieldType) {
+		AssignableFieldTypes assignableFieldType = AssignableFieldTypes.byClass(fieldType);
+		if (assignableFieldType != null) {
+			switch (assignableFieldType) {
```

Comment:
What do you think about moving responsibility from this class to AssignableFieldTypes class. For example by changing enum to Map holding types (interfaces) as key and concrete implementation as values?